src/foam/core/FObject.js

```diff
@@ -824,7 +824,7 @@ foam.CLASS({
 
       // Use comparator of other if it is not an FObject.
       if ( ! foam.core.FObject.isInstance(other) ) {
```

Comment:
The reason I kept it is because `myFObject.compareTo(nonFObject)` may be invoked directly  (without `foam.util.compare()` on the stack). Is that case considered misusing the API? If so, I think it's pretty subtle. If not, then this is still needed.